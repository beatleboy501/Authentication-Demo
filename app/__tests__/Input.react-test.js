import React from 'react';
import Input from '../assets/jsx/components/Input.jsx';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

test('Input renders', () => {
  const component = renderer.create(<Input></Input>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Input can change', () => {
  const component = renderer.create(<Input type="text" label="Username:" valChange={() => 'valChange'} val="val"/>);
  let tree = component.toJSON();
  let onChange = tree.children[2].props.onChange();
  expect(tree).toMatchSnapshot();
  expect(tree.children[0].children[0]).toBe("Username:");
  expect(onChange).toBe('valChange');
});

test('Password can change', () => {
  const component = renderer.create(<Input type="password" label="Password:" valChange={() => 'passwordChange'} val="val"/>);
  let tree = component.toJSON();
  let onChange = tree.children[2].props.onChange();
  expect(tree).toMatchSnapshot();
  expect(tree.children[0].children[0]).toBe("Password:");
  expect(onChange).toBe('passwordChange');
});

test('Interaction', () => {
  const changeSpy = jest.fn();
  const input = shallow(
      <Input name="test" type="text" label="text" valChange={changeSpy} val="val"/>
  );
  input.find("input").simulate("change", "new text");
  expect(changeSpy).toHaveBeenCalledTimes(1);
  expect(changeSpy).toHaveBeenCalledWith("new text")
});
