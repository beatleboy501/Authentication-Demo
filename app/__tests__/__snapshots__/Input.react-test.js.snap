// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input can change 1`] = `
<div>
  <label>
    Username:
  </label>
  <br />
  <input
    onChange={[Function]}
    type="text"
    value="val"
  />
  <br />
</div>
`;

exports[`Input renders 1`] = `
<div>
  <label />
  <br />
  <input
    onChange={undefined}
    type={undefined}
    value={undefined}
  />
  <br />
</div>
`;

exports[`Password can change 1`] = `
<div>
  <label>
    Password:
  </label>
  <br />
  <input
    onChange={[Function]}
    type="password"
    value="val"
  />
  <br />
</div>
`;
